tests/e2e/tests/mixer/mixer_test.go

```diff
@@ -75,56 +68,99 @@ const (
 
 	checkPath  = "/istio.mixer.v1.Mixer/Check"
 	reportPath = "/istio.mixer.v1.Mixer/Report"
+
+	testRetryTimes = 5
 )
 
 type testConfig struct {
 	*framework.CommonConfig
-
 	rulesDir string
 }
 
 var (
-	tc                 *testConfig
+	tc        *testConfig
+	testFlags = &framework.TestFlags{
+		V1alpha1: true,  //implies envoyv1
+		V1alpha3: false, //implies envoyv2
+		Ingress:  true,
+		Egress:   true,
+	}
+	configVersion      = ""
+	ingressName        = "ingress"
 	productPageTimeout = 60 * time.Second
-	rules              = []string{rateLimitRule, denialRule, ingressDenialRule, newTelemetryRule, routeAllRule,
-		routeReviewsVersionsRule, routeReviewsV3Rule, tcpDbRule, bookinfoGateway}
+
+	rulesDir                 string
+	rateLimitRule            string
+	denialRule               string
+	ingressDenialRule        string
+	newTelemetryRule         string
+	routeAllRule             string
+	routeReviewsVersionsRule string
+	routeReviewsV3Rule       string
+	tcpDbRule                string
+	bookinfoGateway          string
+
+	defaultRules []string
+	rules        []string
 )
 
+func init() {
+	testFlags.Init()
+	if len(testFlags.ConfigVersions()) != 1 {
+		panic(fmt.Sprintf("must set exactly one of v1alpha1 or v1alpha3, have %v", testFlags.ConfigVersions()))
+	}
+	configVersion = testFlags.ConfigVersions()[0]
+}
+
+// Setup is called from framework package init().
 func (t *testConfig) Setup() (err error) {
 	defer func() {
 		if err != nil {
 			dumpK8Env()
 		}
 	}()
 
-	var srcBytes []byte
-	for _, rule := range rules {
-		src := util.GetResourcePath(filepath.Join(bookinfoSampleDir, rule+"."+yamlExtension))
-		dest := filepath.Join(t.rulesDir, rule+"."+yamlExtension)
-		srcBytes, err = ioutil.ReadFile(src)
+	rulesDir = "kube"
+	if testFlags.V1alpha3 {
+		rulesDir = "routing"
+		ingressName = "ingressgateway"
+	}
+	rateLimitRule = filepath.Join(rulesDir, "mixer-rule-ratings-ratelimit")
```

Comment:
I would consider write a loop over `rules`, running `filepath.Join` on each of them.

Reply:
A loop over `rules` and `defaultRules`

Reply:
I would consider even move it into `copyRuleToFileSystem`, `getSourcePath` and `getDestinationPath`.

Reply:
Agree with 1 and 2, 3 is probably overkill.

Reply:
OK, agreed.